modules/ensemble/lib/framework/data_context.dart

```diff
@@ -98,6 +98,20 @@ class DataContext implements Context {
     if (_contextMap['device'] == null) {
       _contextMap['device'] = Device();
     }
+
+    // Always set up the listener
+    print('👂 Setting up device update listener...');
```

Comment:
This doesn't make sense. What is it for? DataContext already points to Device class which handle listening for changes.